apps/admin/frontend/src/screens/contest_adjudication_screen.test.tsx

```diff
@@ -1813,43 +1799,35 @@ describe('marginal mark adjudication', () => {
         isVote: true,
       },
     ]);
-    apiMock.expectGetWriteIns({ contestId, cvrId }, []);
-    apiMock.expectGetWriteInCandidates([], contestId);
     apiMock.expectGetCvrContestTag(
       { cvrId, contestId },
       { ...cvrContestTag, isResolved: true }
     );
+    apiMock.expectGetWriteIns({ contestId, cvrId }, []);
+    apiMock.expectGetWriteInCandidates([], contestId);
 
-    apiMock.expectGetCastVoteRecordVoteInfo(
-      { cvrId: cvrId2 },
-      { [contestId]: ['kangaroo'] }
-    );
-    apiMock.expectGetVoteAdjudications({ contestId, cvrId: cvrId2 }, []);
+    // Scroll to next cvr, and scroll back
     apiMock.expectGetWriteIns({ contestId, cvrId: cvrId2 }, []);
+    apiMock.expectGetVoteAdjudications({ contestId, cvrId: cvrId2 }, []);
+    apiMock.expectGetMarginalMarks({ cvrId: cvrId2, contestId }, []);
     apiMock.expectGetCvrContestTag(
       { cvrId: cvrId2, contestId },
-      { ...cvrContestTag, isResolved: true }
+      { ...cvrContestTag, cvrId: cvrId2 }
     );
-    apiMock.expectGetMarginalMarks({ cvrId: cvrId2, contestId }, []);
-    userEvent.click(primaryButton);
-
-    await screen.findByTestId('transcribe:id-175');
+    userEvent.click(getButtonByName('save & next'));
+    await waitForBallotById('id-175');
     userEvent.click(getButtonByName('back'));
 
-    // flags shouldn't be there anymore and captions should remain
-    await screen.findByTestId('transcribe:id-174');
```

Comment:
This line and line 1971 caused the flakiness, as the test didn't wait for the checkboxes, the appearance of which indicates the data populating the digital ballot has fully loaded.